server.go

```diff
@@ -586,6 +546,73 @@ func parseExpansionLevel(raw []string) *ExpansionLevel {
 	return level
 }
 
+// validateAndCoerceRequest validates an incoming request against an OpenAPI
+// schema and does parameter coercion.
+//
+// Firstly, `Content-Type` is checked against the schema's media type, then
+// string-encoded parameters are coerced to expected types (where possible).
+// Finally, we validate the incoming payload against the schema.
+func validateAndCoerceRequest(
```

Comment:
Most of this code is the same, but for cleanliness I've pulled it into its own function.